python/shipDet_conf.py

```diff
@@ -452,18 +408,329 @@ def configure(run, ship_geo):
+            if hasattr(
+                ship_geo.NuTauTT, "scifimat_width"
+            ):  # for backward compatibility
+                NuTauTT.SetSciFiParam(
+                    ship_geo.NuTauTT.scifimat_width,
+                    ship_geo.NuTauTT.scifimat_hor,
+                    ship_geo.NuTauTT.scifimat_vert,
+                    ship_geo.NuTauTT.scifimat_z,
+                    ship_geo.NuTauTT.support_z,
+                    ship_geo.NuTauTT.honeycomb_z,
+                )
+                NuTauTT.SetNumberSciFi(
+                    ship_geo.NuTauTT.n_hor_planes, ship_geo.NuTauTT.n_vert_planes
+                )
+            NuTauTT.SetTargetTrackerParam(
+                ship_geo.NuTauTT.TTX, ship_geo.NuTauTT.TTY, ship_geo.NuTauTT.TTZ
+            )
+            NuTauTT.SetBrickParam(ship_geo.NuTauTarget.CellW)
+            NuTauTT.SetTotZDimension(ship_geo.NuTauTarget.zdim)
+            NuTauTT.SetNumberTT(ship_geo.NuTauTT.n)
+            # method of nutau target that must be called after TT parameter definition
+            NuTauTarget.SetTTzdimension(ship_geo.NuTauTT.TTZ)
+            detectorList.append(NuTauTarget)
+            detectorList.append(NuTauTT)
+
+            # High Precision Tracker
+            if ship_geo.nuTauTargetDesign == 3:
+                tauHpt = ROOT.Hpt(
+                    "HighPrecisionTrackers",
+                    ship_geo.tauHPT.DX,
+                    ship_geo.tauHPT.DY,
+                    ship_geo.tauHPT.DZ,
+                    ROOT.kTRUE,
+                )
+                tauHpt.SetZsize(ship_geo.tauMudet.Ztot)
+                tauHpt.SetDesign(ship_geo.NuTauTarget.Design)
+                if hasattr(
+                    ship_geo.tauHPT, "scifimat_width"
+                ):  # for backward compatibility
+                    tauHpt.SetSciFiParam(
+                        ship_geo.tauHPT.scifimat_width,
+                        ship_geo.tauHPT.scifimat_hor,
+                        ship_geo.tauHPT.scifimat_vert,
+                        ship_geo.tauHPT.scifimat_z,
+                        ship_geo.tauHPT.support_z,
+                        ship_geo.tauHPT.honeycomb_z,
+                    )
+                    tauHpt.SetNumberSciFi(
+                        ship_geo.tauHPT.n_hor_planes, ship_geo.tauHPT.n_vert_planes
+                    )
+                    tauHpt.SetHPTrackerParam(
+                        ship_geo.tauHPT.TX, ship_geo.tauHPT.TY, ship_geo.tauHPT.TZ
+                    )
+
+                tauHpt.SetHPTNumber(ship_geo.tauHPT.nHPT)
+                tauHpt.SetDistanceHPTs(ship_geo.tauHPT.distHPT)
+                if hasattr(ship_geo.tauHPT, "SRDY"):
+                    tauHpt.SetSurroundingDetHeight(ship_geo.tauHPT.SRDY)
+                tauHpt.GetMagnetGeometry(
+                    ship_geo.EmuMagnet.zC, ship_geo.EmuMagnet.Y
+                )
+                tauHpt.GetNumberofTargets(ship_geo.NuTauTarget.target)
+                detectorList.append(tauHpt)
```

Comment:
You just undid @amagnan's change here. Please be more careful when resolving conflicts!